Fix smb-docker fixture when runnig with aufs

alpar-t opened the pull request:
Closes #36073

The problem showed up on debian 8 which uses `aufs` docker storage
driver by default as opposed to `overlay2` used on other distros.
`aufs` does not support acls and thus the failure.
The `--use-ntvfs` option instructs samba not to rely on acls.
From what I can tell this is an implementation detail that should not
affect the tests ( which continue to pass )

elasticmachine commented:
Pinging @elastic/es-core-infra

elasticmachine commented:
Pinging @elastic/es-security

atorok commented:
Since this now has the potential to break the build on Debian and others that default to aufs, I'm going to merge now, and address any PR comments after the fact.

alpar-t committed:
Fix test fixtures on aufs

Closes #36073

The problem showed up on debian 8 which uses aufs docker storage
driver by default as opposed to overlay2 used on other distros.
aufs does not support acls and thus the failure.
The --use-ntvfs option instructs samba not to rely on acls.
From what I can tell this is an implementation detail that should not
affect the tests ( which continue to pass )
